client/galaxy/scripts/layout/menu.js

```diff
@@ -1,396 +1,418 @@
-            title           : 'Shared Data',
-            url             : 'library/index',
-            tooltip         : 'Access published resources',
-            menu            : [{
-                    title   : 'Data Libraries',
-                    url     : 'library/list'
-                },{
-                    title   : 'Histories',
-                    url     : 'history/list_published'
-                },{
-                    title   : 'Workflows',
-                    url     : 'workflow/list_published'
-                },{
-                    title   : 'Visualizations',
-                    url     : 'visualization/list_published'
-                },{
-                    title   : 'Pages',
-                    url     : 'page/list_published'
-            }]
-        });
+    var Collection = Backbone.Collection.extend({
+        model: Backbone.Model.extend({
+            defaults: {
+                visible         : true,
+                target          : '_parent'
+            }
+        }),
+        fetch: function( options ){
+            options = options || {};
+            this.reset();
 
-        //
-        // Lab menu.
-        //
-        options.user_requests && this.add({
-            id              : 'lab',
-            title           : 'Lab',
-            menu            : [{
-                    title   : 'Sequencing Requests',
-                    url     : 'requests/index'
-                },{
-                    title   : 'Find Samples',
-                    url     : 'requests/find_samples_index'
-                },{
-                    title   : 'Help',
-                    url     : options.lims_doc_url
-            }]
-        });
+            //
+            // Chat server tab
+            //
+            var extendedNavItem = new GenericNav.GenericNavView();
+            this.add(extendedNavItem.render());
 
-        //
-        // Visualization tab.
-        //
-        this.add({
-            id              : 'visualization',
-            title           : 'Visualization',
-            url             : 'visualization/list',
-            tooltip         : 'Visualize datasets',
-            disabled        : !Galaxy.user.id,
-            menu            : [{
-                    title   : 'New Track Browser',
-                    url     : 'visualization/trackster',
-                    target  : '_frame'
-                },{
-                    title   : 'Saved Visualizations',
-                    url     : 'visualization/list',
-                    target  : '_frame'
-                },{
-                    title   : 'Interactive Environments',
-                    url     : 'visualization/gie_list',
-                    target  : 'galaxy_main'
+            //
+            // Webhooks
+            //
+            Webhooks.add({
+                url: 'api/webhooks/masthead/all',
+                callback: function(webhooks) {
+                    $(document).ready(function() {
```

Comment:
@guerler if you're talking about `$(document).ready()`, then it is necessary because after plugins have been retrieved, they are being added to `Galaxy.page.masthead`, which is available only after full page load. Therefore, I'm waiting for the pull page load and only then add plugins to masthead.